### PR TITLE
Add actor start and stop audits

### LINKED ISSUE #571: Audit log OpenHIM application start

### PR DESCRIPTION
**Pull request by rcrichton**
Closes #571

### CODE REVIEW

**Comment by rcrichton** [comment]
@devcritter could you have a look at this one as well. The problem here is that the startup and shutdown audit will get logged once per cluster worker. I can't find a better way to do this and technically that may be the correct thing to do. What do you think?

**Comment by devcritter** [comment]
@rcrichton cool, looks good to me - I just haven't had a chance to look into the audit library yet; but I'll make any comments regarding in the other PR so that it's in one place.

I'm not sure what the best practice is wrt ATNA, but it seems alright to me to have every worker audit - rather more than less I think. Besides if someone clustered over mutliple servers - it would anyway have audited multiple times - which seems correct to me.

Otherwise all good :+1: although some of the issues in the other PR might also apply here...

**Comment by devcritter** [comment]
Oh I almost forgot: On the ticket there's a discussion about also auditing mediator startups when they register. Do you want to take this as part of this PR?

**Comment by rcrichton** [comment]
Ok cool. I'l take a look at adding mediator registration auditing. That shouldn't be too hard.

**Comment by rcrichton** [comment]
@devcritter Ok, I've made the changes and added start auditing for mediators as well.

File: src/api/mediators.coffee [hunk 1]
```diff
@@ -53,6 +55,18 @@ exports.addMediator = ->
...
   try
     mediator = this.request.body
+
+    if mediator?.endpoints?[0]?.host?
+      mediatorHost = mediator.endpoints[0].host
+    else
+      mediatorHost = 'unknown'
+
+    # audit mediator start
+    audit = atna.appActivityAudit true, mediator.name, mediatorHost, 'root'
```
Comment (devcritter) [inline src/api/mediators.coffee]: to be fair, we don't actually know that it's the root user. Not sure if there's a way to pick up the user dynamically - couldn't see anything from a quick search - maybe rather call it `system` instead? just to make it a clearer

**Comment by devcritter** [comment]
Looks good. Just a minor comment about using the root user if you agree, but otherwise :+1:

File: src/api/mediators.coffee (hunk 1 above)
Comment (rcrichton) [inline src/api/mediators.coffee]: Yeah, I just put root there for lack of something better. `system` does sound better. I'l change it.

**Pull request merged**